src/callbacks.c

```diff
@@ -1,4 +1,12 @@
 #include "curl-common.h"
+#include "utils.h"
+
+int default_callback_progress(void *dummy,
+                              double dltotal, double dlnow,
+                              double ultotal, double ulnow)
+{
+  return !(R_ToplevelExec(check_interrupt_fn, NULL));
```

Comment:
Actually my experience of long jumping out of the progress callback can leave libcurl in an unstable ambiguous state.